src/testRunner/unittests/helpers/tsserver.ts

```diff
@@ -154,6 +155,10 @@ export class TestSession extends ts.server.Session {
         request.type = "request";
         return this.executeCommand(request);
     }
+
+    protected override shouldDoRegionCheck(_file: ts.server.NormalizedPath): boolean {
+        return true;

```

Comment:
I would also add option in `TestSessionOptions` to `disableRegionDiagnostics?: boolan` and use that here to return false if enabled. So you can add tests when region diagnsotics are enabled but we skip them

Reply:
I would also write it as below to ensure we are running logic for the real session here as:
`return super.shouldDoRegionCheck(file) && !this.disableRegionDiagnostics;`

Reply:
I think the point of the this is to bypass the actual logic, since the test files will not be large enough to make the actual session implementation do the region check. I'll add an option to explicitly enable this bypassing, always on region check.